### PR TITLE
refactor: replace linkify-markdown

### PR DESCRIPTION
**Pull request by viceice**
## Changes:

Removes `linkify-markdown` to fix [GHSA-w5p7-h5w8-2hfq](https://github.com/advisories/GHSA-w5p7-h5w8-2hfq). Also allows us to use newer `remark` and `remark-github`.

Not sure why there are snapshot changes. 😕 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: patches/vfile-message+2.0.4.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/node_modules/vfile-message/types/index.d.ts b/node_modules/vfile-message/types/index.d.ts
+index d217843..31c853a 100644
+--- a/node_modules/vfile-message/types/index.d.ts
++++ b/node_modules/vfile-message/types/index.d.ts
+@@ -6,7 +6,7 @@ declare namespace vfileMessage {
+   /**
+    * Create a virtual message.
+    */
+-  interface VFileMessage extends Error {
++  interface VFileMessage extends Omit<Error, 'location'> {
```
Comment (viceice) [inline patches/vfile-message+2.0.4.patch:10]: https://github.com/vfile/vfile-message/issues/6 😕

Comment (viceice) [reply patches/vfile-message+2.0.4.patch:10]: alternate is to use `skipLibCheck` which disables typescript checks on external libs 😕

Comment (JamieMagee) [reply patches/vfile-message+2.0.4.patch:10]: I would prefer the new `2.x` release of `vfile-message`. It looks like the owner of `unified` is holding off for their next version bump: https://github.com/unifiedjs/unified/issues/121

Comment (JamieMagee) [reply patches/vfile-message+2.0.4.patch:10]: Yeah, I noticed that as well 😑

Comment (viceice) [reply patches/vfile-message+2.0.4.patch:10]: So only patch-package or skiplibcheck are working here.

Comment (viceice) [reply patches/vfile-message+2.0.4.patch:10]: @JamieMagee Some progress at https://github.com/vfile/vfile-message/issues/6 maybe we can fix types external before or after this soon.

**Comment by viceice** [comment]
`remark-github` github has now build-in types 🎉 

https://github.com/remarkjs/remark-github/pull/29

File: patches/vfile-message+2.0.4.patch (hunk 1 above)
Comment (viceice) [reply patches/vfile-message+2.0.4.patch]: fixed

**Comment by viceice** [comment]
Ready for another review. package patching no longer nessesary. 🎉

**Comment by rarkins** [comment]
Can we run it on our example repos to see if/how it changes markdown compared to the existing solution?

**Comment by viceice** [comment]
> Can we run it on our example repos to see if/how it changes markdown compared to the existing solution?

this pr: https://github.com/viceice-tests/renovate-changelog1

current main: https://github.com/viceice-tests/renovate-changelog2

They look same to me. Any other dependency to add and check?

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**